internal/controller/scheduler/imagePullBackOffWatcher.go

```diff
@@ -147,3 +152,12 @@ func shouldCancel(containerStatus *v1.ContainerStatus) bool {
 	return containerStatus.State.Waiting != nil &&
 		containerStatus.State.Waiting.Reason == "ImagePullBackOff"
 }
+
+// We don't want to kill jobs because of sidecars error.
+// The name "system container" is subject to more debate.
+func isSystemContainer(containerStatus *v1.ContainerStatus) bool {
+	name := containerStatus.Name
+	// This will arguably cause false positive, but we will replace this soon.
+	matched, _ := regexp.MatchString(`container-\d+`, name)
```

Comment:
It's probably wise to capture error from 
`agent`, `copy-agent` and `checkout` too. But I want to double check if there are already pre-existing error detection logic for these error path.